### DISCUSSION TITLE
Overlay image name onto image, through folder and sub-folders

### DISCUSSION THREAD

**Discussion by BavisiDhiren** [ImageMagick/ImageMagick#2269] [Help]
Hi all,

I've never used any sort of CLI before - hence apologies in advance if this comes across non-sensical.

I'm trying to take the image name and overlay it to the bottom center (maybe bottom left), of all images through a folder and sub-folder. The only change is the "watermark"-like effect of the filename without the extension, onto the image.

So far, I've been advised this-

`    convert input.jpg -gravity South -annotate 0 '%t' output.jpg`

and was told that it needs an additional loop and something to handle the filenames. Is this the right approach? Whats a good way to do this? I tried googling and couldn't figure it out :/

Any and all advices are appreciated

**Comment by snibgo** [comment]
You don't say what your platform is (eg bash or Windows). In Windows, I normally use a "for" loop to process folders and files within folders. Bash also has a "for" command, but it works differently.

Bash and Windows have interactive help, which give the syntax and short explanations. The web has many many pages that give more detail and examples.

**Reply by BavisiDhiren** [reply]
Sorry, it is windows platform - didn't realize the commands were different

**Reply by fmw42** [reply]
See mogrify at https://imagemagick.org/Usage/basics/#mogrify to process a whole folder of images with the same commands
